editor/editor_file_system.cpp

```diff
@@ -1624,14 +1786,41 @@ void EditorFileSystem::_update_files_icon_path(EditorFileSystemDirectory *edp) {
 }
 
 void EditorFileSystem::_update_script_classes() {
+	if (update_script_paths.is_empty()) {
```

Comment:
You could add the same early return in `_update_scene_groups()`.
I was going to open a PR for that, but it would overlap with yours.

Reply:
Done!